hibernate-core/src/test/java/org/hibernate/test/proxy/narrow/ProxyNarrowingTest.java

```diff
@@ -84,4 +104,55 @@ public void testNarrowedProxyIsInitializedIfOriginalProxyIsInitialized() {
 		}
 	}
 
+	@Test
+	public void testHHH11280() {
+		Session s = openSession();
+		try {
+			// Populate the database
+			{
+				s.beginTransaction();
+				UserConfig config = new UserConfig();
+				AdvancedUserDetail detail = new AdvancedUserDetail();
+				AdvancedUser advUser = new AdvancedUser();
+				detail.setId(9L);
+				detail.setStatusCode("A");
+				detail.setAdvancedUser(advUser);
+				advUser.setId(11L);
+				advUser.setType(106);
+				advUser.setCurrentDetail(detail);
+				config.setId(10L);
+				config.setUser(advUser);
+				s.persist(advUser);
+				s.persist(detail);
+				s.persist(config);
+				s.flush();
+				s.getTransaction().commit();
+			}
+
+			// Get on with the test.
+			s = openSession();
+			DetachedCriteria configCriteria = DetachedCriteria.forClass(UserConfig.class);
+			configCriteria.add(Restrictions.eq("id", new Long(10)));
+			final List<UserConfig> configRows = configCriteria.getExecutableCriteria(s).list();
+			UserConfig config = configRows.get(0);
+			User configUser = config.getUser();
+			Assert.assertNotNull(configUser);
+	
+			try {
+				DetachedCriteria userCriteria = DetachedCriteria.forClass(User.class);
+				userCriteria.add(Restrictions.eq("id", new Long(11)));
+				final List<User> userRows = userCriteria.getExecutableCriteria(s).list();
```

Comment:
Before the fix in this commit, `user` in the next line would be the same as `configUser` as it should be.

Using the fix in this commit, `user` and `configUser` are different objects.

As mentioned above, `configUser` will still contain a `Session` reference even after the `Session` is closed.